cli/src/cli_util.rs

```diff
@@ -2059,12 +2059,27 @@ See https://jj-vcs.github.io/jj/latest/working-copy/#stale-working-copy \
             .map(|commit_id| tx.repo().store().get_commit(commit_id))
             .transpose()?;
 
+        // Determine if we need to acquire the WC lock:
+        // - If we're exporting Git HEAD (colocated repo with new WC commit)
+        // - OR if we're updating the working copy files
+        let need_wc_lock = {
+            let git_head_export =
+                self.working_copy_shared_with_git && maybe_new_wc_commit.is_some();
+            let wc_update = self.may_update_working_copy && maybe_new_wc_commit.is_some();
+            git_head_export || wc_update
+        };
+
+        let mut locked_ws = if need_wc_lock {
+            Some(self.workspace.start_working_copy_mutation()?)
+        } else {
+            None
+        };
```

Comment:
nit: Can combine these since `need_wc_lock` isn't needed later:

```rust
        let mut locked_ws = {
            let git_head_export =
                self.working_copy_shared_with_git && maybe_new_wc_commit.is_some();
            let wc_update = self.may_update_working_copy && maybe_new_wc_commit.is_some();
            let need_wc_lock = git_head_export || wc_update;
            if need_wc_lock {
                Some(self.workspace.start_working_copy_mutation()?)
            } else {
                None
            }
        };
```

Feel free to also inline `need_wc_lock`.